### PR TITLE
Add Text Concatenated Variadic Decorator

### PR DESCRIPTION
**Pull request by haspadar**
- Added Text Concatenated decorator providing variadic sugar over Joined with empty separator
- Added tests covering zero, one, two, three parts and multibyte content

Part of #209

## Summary by CodeRabbit

* **New Features**
  * Introduced a text concatenation capability that seamlessly combines multiple text parts into a single value, preserving order and character integrity throughout the process.

* **Tests**
  * Added comprehensive test coverage to verify concatenation behavior, including proper handling of multibyte characters, empty input scenarios, and edge cases with single text inputs.

[Review Change Stack](https://app.coderabbit.ai/change-stack/haspadar/primus/pull/211?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/Text/Concatenated.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+final readonly class Concatenated extends TextEnvelope
+{
+    /**
+     * Ctor.
+     *
+     * @param Text ...$parts The texts to concatenate.
+     */
+    public function __construct(Text ...$parts)
+    {
+        parent::__construct(new Joined('', array_values($parts)));
```
[the remark this answers is not in the document]
Comment (haspadar) [reply src/Text/Concatenated.php]: Applied the named constant for the separator. Kept `array_values($parts)`: PHP variadic gives `array<int, Text>`, not `list<Text>`, and `Joined::$parts` is typed as `list<Text>`. Without the narrowing call PHPStan fails with `argument.type` — `array_values` is a static-narrowing helper here, not eager computation in the EO sense.
